### PR TITLE
Fixed issue when IN query statement didn't work with Long values (cherry pick from main)

### PR DESCRIPTION
**Pull request by rinatmini**
# Description
Related issue: #36209 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by rinatmini** [comment]
run /azp java - spring - cosmos

**Comment by rinatmini** [comment]
run /azp java - spring - cosmos

File: sdk/spring/azure-spring-data-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue with IN statement in annotated queries not working for `Long` data type - See [PR 36249](https://github.com/Azure/azure-sdk-for-java/pull/36249).
```
Comment (kushagraThapar) [inline sdk/spring/azure-spring-data-cosmos/CHANGELOG.md:10]:
```suggestion
* Fixed an issue with IN statement in annotated queries not working for `Long` data type - See [PR 36267](https://github.com/Azure/azure-sdk-for-java/pull/36267).
```

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @rinatmini

**Pull request merged**